Update ZHA documentation with tips on improving Zigbee network range

Hedda opened the pull request:
## Proposed change

Update ZHA docs with best practice guidelines for improving Zigbee network range and coverage by following some general tips.

Suggest adding this to [ZHA integration](https://www.home-assistant.io/integrations/zha) troubleshooting section as questions and issues or problems related to range or coverage as well as Zigbee network stability with lost connections due to common known sources of interference are very common today. 

FYI, as the ZHA integration (and other Zigbee implementations) are getting popular, more end-user beginners and new users of Zigbee devices are posting in the Home Assistant community forums about issues/problems related to Zigbee range and stability.

https://community.home-assistant.io/c/configuration/zha/48

https://community.home-assistant.io/c/configuration/zigbee/39

Note! Please remember that while users of other Zigbee implementations (like ex. Zigbee2MQTT and deCONZ) can be referred to extensive documentation on their respective websites, the [ZHA integration](https://www.home-assistant.io/integrations/zha) have no other website/pages with documentation.

These suggestions will therefore advise ZHA users on relatively simple actions they can and should take if they wish to greatly improve signal range and mesh network coverage or lessen possible signal interference due to electromagnetic interference (a.k.a. radio interference). All of these essential actions to set up a solid Zigbee network can easily and relatively quickly be solved by the users themselves. Also following the recommendations as a new user and Zigbee beginners should a lot of people much frustration as poor signal reception quality can cause intermittent connection issues which can be a pain to troubleshoot.

The recommendations are largely indirectly based on posts in different communities with recommendations from experienced Zigbee developers (like Koen Kanters a.k.a. "Koenkk" of Zigbee2MQTT fame and Omer Kilic a.k.a. "omerk" who designed the open-source zzh adapter) plus several longer Zigbee range discussions threads found all around in the Home Assistant / ZHA / Zigpy, Zigbee2MQTT, Hubitat, openHAB, and Domoticz community forums with users tips on how-to improve network range and stability and other support channels.

For example, the most important advice posted in the forums is for users to use a long USB-extension cable for their Zigbee Coordinator and to also buy more "Zigbee Router" devices (normally AC mains-powered Zigbee devices) to their home because they will route/repeat the Zigbee communication and as such extend the network range. Also, more common advice is upgrading firmware on both Zigbee Coordinator as well as Zigbee devices, or just buying a better Zigbee Coordinator adapter if users are using ones that are based on an obsolete chip and/or one for which the firmware is no longer maintained by the manufacturer.

Below are just a few direct references with questions and answers related to connectivity posted in Home Assistant community forums, where most of these topics could probably have been resolved quicker if those or similar troubleshooting tips in ZHA docs:

https://community.home-assistant.io/t/advice-on-zigbee-range-extending/175882
https://community.home-assistant.io/t/zha-is-not-stable-please-help/173203/
https://community.home-assistant.io/t/building-out-my-zha-network/196709
https://community.home-assistant.io/t/extend-zigbee-range-coverage-to-reach-distant-xiaomi-aqara-door-sensor/110329/
https://community.home-assistant.io/t/migrating-from-hue-hub-to-conbee-questions-thoughts-about-coverage-and-extending-range/236569/
https://community.home-assistant.io/t/best-device-to-act-as-a-zha-zigbee-repeater/103904
https://community.home-assistant.io/t/understanding-the-zigbee-landscape/94847/
https://community.home-assistant.io/t/i-am-getting-so-sick-of-zigbee-random-disconnections-basically-every-time-i-reboot/226372/
https://community.home-assistant.io/t/solved-zha-not-working-with-conbee-ii/370164/
https://community.home-assistant.io/t/unable-to-get-home-decorator-hampton-bay-zigbee-fan-to-pair/193657
https://community.home-assistant.io/t/itead-zigbee-3-0-usb-dongle-stick-only-cost-7-and-is-based-on-silicon-labs-efr32mg21/271144
https://community.home-assistant.io/t/difficulty-connecting-zigbee-devices/359823/
https://community.home-assistant.io/t/conbee-ii-woes-only-works-when-holding-it/366207/
https://community.home-assistant.io/t/zigbee-conbee-and-wifi-interference/112945
https://community.home-assistant.io/t/usb3-0-radio-frequency-interference/288695
https://community.home-assistant.io/t/zigbee-errors/292478/
https://community.home-assistant.io/t/almost-at-breaking-point-with-zigbee/283768/
https://community.home-assistant.io/t/reliability-issues-and-pairing-issues-even-after-trying-multiple-coordinators-and-following-all-advice-i-could-find/384196/

https://www.reddit.com/r/homeassistant/comments/eym1h8/state_of_zigbee_in_2020_zha_z2m_deconz/
https://www.reddit.com/r/homeassistant/comments/czi8lx/zigbee_range/
https://www.reddit.com/r/homeassistant/comments/lfu1yh/why_is_my_zigbee_network_so_screwy/
https://www.reddit.com/r/homeassistant/comments/lo20sh/is_zigbee_reliable_struggle/
https://www.reddit.com/r/homeassistant/comments/k1xtdh/having_problems_with_zigbee/
https://www.reddit.com/r/homeassistant/comments/oqja47/should_i_continue_with_zigbee/
https://www.reddit.com/r/homeassistant/comments/s2b77k/facepalm_zigbee_really_annoying_me_for_months/
https://www.reddit.com/r/homeassistant/comments/qmyvrb/home_assistant_zigbee_stability_issues_zha/
https://www.reddit.com/r/homeassistant/comments/q0xpgd/having_trouble_with_my_zigbee_suddenly_today_all/

Well researched with indirectly inspirations from basic Zigbee troubleshooting tips in other communities and other references:

https://www.zigbee2mqtt.io/advanced/zigbee/02_improve_network_range_and_stability.html#usb-based-adapter
https://www.zigbee2mqtt.io/guide/faq/
https://docs.hubitat.com/index.php?title=How_to_Build_a_Solid_Zigbee_Mesh
https://community.hubitat.com/t/xiaomi-aqara-devices-pairing-keeping-them-connected/623
https://home-assistant-guide.com/2020/10/29/choose-your-zigbee-channel-wisely/
https://electrolama.com/radio-docs/troubleshooting/
https://gadget-freakz.com/how-to-improve-your-zigbee-network/
https://nts.com/services/education/articles/zigbee-best-practices/
https://zigbeealliance.org/news_and_articles/step-by-step-guide-getting-started-with-zigbee/
https://www.digikey.com/en/articles/tips-for-optimizing-performance-and-energy-use-of-zigbee-radios
https://www.control4.com/docs/product/zigbee/best-practices/english/revision/B/
https://www.metageek.com/training/resources/zigbee-wifi-coexistence/
https://community.smartthings.com/t/faq-networking-and-reducing-channel-interference-between-wifi-and-zigbee/40159/3
https://notenoughtech.com/featured/a-simple-trick-to-reduce-zigbee-coordinator-wifi-interference/
https://phoscon.de/en/support
https://www.usb.org/sites/default/files/327216.pdf (USB 3.0* Radio Frequency Interference Impact on 2.4 GHz Wireless Devices)
https://www.unit3compliance.co.uk/2-4ghz-intra-system-or-self-platform-interference-demonstration/

https://www.youtube.com/watch?v=1dcAXkJxzcY&ab_channel=digiblurDIY
https://www.youtube.com/watch?v=t-gw7kURXCk&t=2s&ab_channel=HomeAutomationGuy
https://www.youtube.com/watch?v=UmpDXc3cXbU&ab_channel=EverythingSmartHome

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: NA
- Link to parent pull request in the Brands repository: NA
- This PR fixes or closes issue: NA

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

Hedda committed:
Update zha.markdown with tips on improving Zigbee network range

Update zha.markdown with tips on improving Zigbee network range and coverage.

Hedda commented:
Possible to get this reviewed? Again these answers some of the most asked questions in Home Assistant community forum for ZHA.

frenck commented:
While I do get this is being added, it is really really hard to read. Bot in markdown and in the actual result on the website.

<img width="1526" alt="CleanShot 2022-02-19 at 00 21 25@2x" src="https://user-images.githubusercontent.com/195327/154773926-e73106ee-016c-444b-8611-0fb28b432e91.png">

Before this can be considered, it really needs to become more readable.

Hedda commented:
> Before this can be considered, it really needs to become more readable.

Are there maybe some volunteering technical writers/editors in the community who are willing to help with that here?

Help is wanted as I am not good at writing documentation shorter myself, (and I am also not a native English speaker).

I can write more and longer on the topic but find it really hard to write less that still convey all the same information.

Hedda commented:
Rather than wait another six months, is it not better to add this 'as-is' now and then let someone rewrite parts of it in a new PR?

frenck commented:
> Rather than wait another six months, is it not better to add this 'as-is' now and then let someone rewrite parts of it in a new PR?

I dunno, you could also try to improve on it?

Corecii commented:
This is hard to make concise because there's *a lot* of information here.

Here's my second attempt:

<details><summary>Second attempt (click to expand)</summary>

Poor signal quality and signal interference can cause significant degradation in Zigbee network performance. Here are some tips for optimizing signal quality and coverage. Following all of these tips will typically resolve most range and transmission issues.

Common factors of Zigbee connectivity:
* 2.4 GHz RF interference (from routers and other electronics)
* Conflicting Zigbee and WiFi channels
* USB 3 RF interference
* Bad coordinator placement
* Bad coordinator orientation
* Too many obstacles between the coordinator or router and end devices
* Too few routers / repeaters between the coordinator and end devices
* Devices not paired in their final locations
* End devices paired *before* routers are paired
* Underpowered coordinator
* Underpowered coordinator antenna
* Bad / outdated / incompatible firmware on the coordinator
* Hardware issue in the connection between the coordinator and the host computer
* Issues in the host software

For these reasons, we suggest:
* Test the network with devices in line-of-sight of the coordinator
* Pair your end devices *after* pairing your router / repeater devices
* Put your Zigbee network on the channel that interferes the least with your neighbors' WiFi
* Put your WiFi network on the channel that interferes the least with your Zigbee network
* Only use USB 2 ports to connect your coordinator
* Use a *long* USB 2 extension cable to place and orient your coordinator properly, away from interference
* Use the best firmware for your coordinator according to the community of that coordinator
* Try a different host computer
* Try a powerful coordinator with an external antenna
* Try a coordinator that connects over Ethernet instead of USB
* Try different Zigbee host software

Resources:
* [ZigBee and Wi-Fi Coexistence](https://www.metageek.com/training/resources/zigbee-wifi-coexistence.html)
* [put a link to repeaters here]
* [put some other link here]

</details>

<details><summary>Source</summary>

```markdown
Poor signal quality and signal interference can cause significant degradation in Zigbee network performance. Here are some tips for optimizing signal quality and coverage. Following all of these tips will typically resolve most range and transmission issues.

Common factors of Zigbee connectivity:
* 2.4 GHz RF interference (from routers and other electronics)
* Conflicting Zigbee and WiFi channels
* USB 3 RF interference
* Bad coordinator placement
* Bad coordinator orientation
* Too many obstacles between the coordinator or router and end devices
* Too few routers / repeaters between the coordinator and end devices
* Devices not paired in their final locations
* End devices paired *before* routers are paired
* Underpowered coordinator
* Underpowered coordinator antenna
* Bad / outdated / incompatible firmware on the coordinator
* Hardware issue in the connection between the coordinator and the host computer
* Issues in the host software

For these reasons, we suggest:
* Test the network with devices in line-of-sight of the coordinator
* Pair your end devices *after* pairing your router / repeater devices
* Put your Zigbee network on the channel that interferes the least with your neighbors' WiFi
* Put your WiFi network on the channel that interferes the least with your Zigbee network
* Only use USB 2 ports to connect your coordinator
* Use a *long* USB 2 extension cable to place and orient your coordinator properly, away from interference
* Use the best firmware for your coordinator according to the community of that coordinator
* Try a different host computer
* Try a powerful coordinator with an external antenna
* Try a coordinator that connects over Ethernet instead of USB
* Try different Zigbee host software

Resources:
* [ZigBee and Wi-Fi Coexistence](https://www.metageek.com/training/resources/zigbee-wifi-coexistence.html)
* [put a link to repeaters here]
* [put some other link here]
```

</details>

Here's my first attempt, if it helps:

<details><summary>First attempt (click to expand)</summary>

Poor signal quality and signal interference can cause significant degradation in Zigbee network performance. Here are some tips for optimizing signal quality and coverage. Following all of these tips will typically resolve most range and transmission issues.

* Test the network with a small number of devices to minimize external variables.
  For an initial test your coordinator should have line-of-sight to connected Zigbee devices. Once that works consistently, *re-pair* your devices in more realistic test locations or in their final locations.
* Add routers (repeaters) between the coordinator and end devices.
  Zigbee is a mesh network. If you're seeing issues connecting to far away devices or devices with walls between them and the coordinator then a router / repeater may solve your issues.
  **Reset your end device after adding a router / repeater.** Most devices will not connect to a router / repeater unless initially paired with the router / repeater in place.
  Most (but not all) mains-powered devices will act as a router / repeater. Battery-powered devices are *not* routers / repeaters.
* Pair your devices *after* pairing all routers / repeaters.
* Pair your devices in their final location.
  Zigbee builds its mesh according to where devices are *paired* at. Most devices do not re-build their connections when moved, so the mesh will often not adapt to you moving a device after pairing it.
* Place your Zigbee coordinator far from other electronic devices.
  Interference from 2.4 GHz Wifi is a major source of Zigbee issues.
  Other equipment can emit RF noise in the 2.4 GHz range, like USB 3 ports on computers.
* Place your Zigbee coordinator central to your house or your Zigbee devices.
  Avoid walls and other obstacles between your coordinator and devices, where possible.
  Try initially testing with devices in line-of-sight to each other.
* Try different orientations of your Zigbee coordinator's antenna.
  If your coordinator is an all-in-one device like a USB stick, re-orient the whole device.
* Avoid USB 3.
  USB 3 can emit RF noise in the 2.4 GHz range, which interferes with Zigbee.
  Ensure you're using a USB 2 port, and try different ports to see if one performs better.
  A USB 2 extension cable may help eliminate some USB 3 RF noise.
* Use a *long* USB 2 extension cable *or* use a Zigbee router that connects over Ethernet.
  A long cable allows you to avoid interference and orient your coordinator optimally.
* Ensure 2.4 GHz WiFi and Zigbee are not operating on the same RF channels.
  * See [ZigBee and Wi-Fi Coexistence](https://www.metageek.com/training/resources/zigbee-wifi-coexistence.html) for a a detailed description of how Zigbee and WiFi can interfere.
  * Use a WiFi scanning app to see which channels your neighbors are using. Place your Zigbee network on the Zigbee channel with the least interference from your neighbors.
  * Place your 2.4 GHz WiFi on a channel that does not overlap your Zigbee network.
  * Changing your Zigbee channel require re-pairing all Zigbee devices.
* Update the firmware of your Zigbee coordinator to the most-compatible version.
  This is typically the most-recent version, but sometimes older versions have better compatibility. Check community resources for your Zigbee router first.
* Try a different computer, or try a Zigbee router that connects over Ethernet.
  Rarely, an issue with the host computer may be at fault.
  Cut the host computer out of the equation by testing with a different computer or using an Ethernet coordinator.
* If your coordinator has a removable antenna, you can try a directional high-gain antenna.
  This is only necessary for covering long distances or significant obstacles. If you are trying to resolve fundamental issues with close devices you have line-of-sight too, then something else is the cause.
* Buy a more powerful Zigbee coordinator.
  Zigbee coordinators with external antennas give you the best power, range, and flexibility.
  Zigbee coordinators that work over Ethernet give you the most flexibility and eliminate hardware connection issues between the coordinator and the host computer.

</details>

<details><summary>Source</summary>

```markdown
Poor signal quality and signal interference can cause significant degradation in Zigbee network performance. Here are some tips for optimizing signal quality and coverage. Following all of these tips will typically resolve most range and transmission issues.

* Test the network with a small number of devices to minimize external variables.
  For an initial test your coordinator should have line-of-sight to connected Zigbee devices. Once that works consistently, *re-pair* your devices in more realistic test locations or in their final locations.
* Add routers (repeaters) between the coordinator and end devices.
  Zigbee is a mesh network. If you're seeing issues connecting to far away devices or devices with walls between them and the coordinator then a router / repeater may solve your issues.
  **Reset your end device after adding a router / repeater.** Most devices will not connect to a router / repeater unless initially paired with the router / repeater in place.
  Most (but not all) mains-powered devices will act as a router / repeater. Battery-powered devices are *not* routers / repeaters.
* Pair your devices *after* pairing all routers / repeaters.
* Pair your devices in their final location.
  Zigbee builds its mesh according to where devices are *paired* at. Most devices do not re-build their connections when moved, so the mesh will often not adapt to you moving a device after pairing it.
* Place your Zigbee coordinator far from other electronic devices.
  Interference from 2.4 GHz Wifi is a major source of Zigbee issues.
  Other equipment can emit RF noise in the 2.4 GHz range, like USB 3 ports on computers.
* Place your Zigbee coordinator central to your house or your Zigbee devices.
  Avoid walls and other obstacles between your coordinator and devices, where possible.
  Try initially testing with devices in line-of-sight to each other.
* Try different orientations of your Zigbee coordinator's antenna.
  If your coordinator is an all-in-one device like a USB stick, re-orient the whole device.
* Avoid USB 3.
  USB 3 can emit RF noise in the 2.4 GHz range, which interferes with Zigbee.
  Ensure you're using a USB 2 port, and try different ports to see if one performs better.
  A USB 2 extension cable may help eliminate some USB 3 RF noise.
* Use a *long* USB 2 extension cable *or* use a Zigbee router that connects over Ethernet.
  A long cable allows you to avoid interference and orient your coordinator optimally.
* Ensure 2.4 GHz WiFi and Zigbee are not operating on the same RF channels.
  * See [ZigBee and Wi-Fi Coexistence](https://www.metageek.com/training/resources/zigbee-wifi-coexistence.html) for a a detailed description of how Zigbee and WiFi can interfere.
  * Use a WiFi scanning app to see which channels your neighbors are using. Place your Zigbee network on the Zigbee channel with the least interference from your neighbors.
  * Place your 2.4 GHz WiFi on a channel that does not overlap your Zigbee network.
  * Changing your Zigbee channel require re-pairing all Zigbee devices.
* Update the firmware of your Zigbee coordinator to the most-compatible version.
  This is typically the most-recent version, but sometimes older versions have better compatibility. Check community resources for your Zigbee router first.
* Try a different computer, or try a Zigbee router that connects over Ethernet.
  Rarely, an issue with the host computer may be at fault.
  Cut the host computer out of the equation by testing with a different computer or using an Ethernet coordinator.
* If your coordinator has a removable antenna, you can try a directional high-gain antenna.
  This is only necessary for covering long distances or significant obstacles. If you are trying to resolve fundamental issues with close devices you have line-of-sight too, then something else is the cause.
* Buy a more powerful Zigbee coordinator.
  Zigbee coordinators with external antennas give you the best power, range, and flexibility.
  Zigbee coordinators that work over Ethernet give you the most flexibility and eliminate hardware connection issues between the coordinator and the host computer.
```

</details>

I think the best way to make this more concise is to slim each significant point down to a single sentence, then either provide a link to more information or let the user use it as a starting point for their own search.

There's too much information to put it all here at once. Presenting the user a concise summary of all possible problems as a starting point is valuable as it lets users easily scan for problems and it's really the only way to hold the whole set of problems and solutions in your head.

Feel free to take either of my attempts above and expand on them.

(I came from one of the linked posts. Having all this information available would have helped me, but I agree it needs to be made more concise!)

Hedda commented:
> there's _a lot_ of information here

While it could wait I would actually like to add another bullet point about shielding your computer and USB devices or peripherals:

**_Shield your computer and USB devices or peripherals connected to computers as well as other appliances or cables/wires located close to your Zigbee Coordinator. As already established, Zigbee radio can be very susceptible to electromagnetic interference (EMI, also called radio-frequency interference or RFI), and one very common source that generates electromagnetic fields is unshielded computers and unshielded peripherals so make sure that you use for example enclosures for external hard drives and single-board-computers that are made of metal instead of plastic. Especially known to have a huge interference impact on wireless devices using radio frequency like Zigbee radios are USB 3.0 devices/peripherals (for more details on that read white paper report on the subject by Intel at https://www.usb.org/sites/default/files/327216.pdf)._**

The reason for this is seeing more and more community posts with Zigbee interference problems caused by electromagnetic fields.

frenck commented:
Ok, I'm stumbling on this PR again (during my pre-release preparation rounds). Either this PR works on the contents, or I'll consider it stale and close it.

I really appreciate any improvement to the documentation to the documentation, but as said before above; this is really unreadable. 

If you have no intention to improve it, please close it. Thanks 👍

Hedda commented:
> Ok, I'm stumbling on this PR again (during my pre-release preparation rounds). Either this PR works on the contents, or I'll consider it stale and close it.
> 
> I really appreciate any improvement to the documentation to the documentation, but as said before above; this is really unreadable.
> 
> If you have no intention to improve it, please close it. Thanks 👍

I can edit it to try to shorten or separate some sentences and add more full stops. But I then hope you add it as-is as IMHO it is better to add something on this as a baseline that others can work from and edit than to have nothing at all as you have now.

frenck commented:
> is better to add something on this as a baseline that others can work from 

We don't agree on this being a baseline. Again, its unreadable.

Hedda committed:
Improving Zigbee network range tips for zha.markdown

Improving Zigbee network range tips for ZHA in zha.markdown

Split and converted previous submission to shorter bullet points.

Hedda commented:
OK, I have now tried to fully break up or split and converted the previous submission attemt into much shorter bullet points:

https://github.com/home-assistant/home-assistant.io/pull/18864/commits/970295a277e8f01d3ee39eeeaacf453625b988d3

As mentioned by @Corecii it is hard to make this first attempt shorter and concise because there is still a lot of information.

### Improving Zigbee network range

Poor signal quality or interference can lead to transmission problems and show symptoms such as or errors when sending and/or receiving Zigbee messages that will cause significant degradation in Zigbee network performance. Below is some fundamental background information and tips for a Zigbee setup starting point to achieve better signal quality, coverage and range.

Following these essential optimization tips will typically resolve many known issues caused by a bad setup of Zigbee Coordinator adapter and Zigbee network mesh or Zigbee devices. At the very least taking these actions should improve most message transmitting and receiving issues caused by not knowing the basics needed to workaround related Zigbee limitations.

#### Understanding common causes of problems in Zigbee setups

- Zigbee Coordinator adapter hardware:
  - Poor placement of the Zigbee Coordinator and/or wrong orientation of Zigbee Coordinator antenna.
  - Poor / outdated / incompatible Zigbee Coordinator firmware on the Zigbee Coordinator adapter.
  - Old, outdated or underpowered Zigbee Coordinator hardware or poor Zigbee Coordinator antenna.
- Zigbee Coordinator and Zigbee devices sensitive to all types of RFI/EMI/EMF interference:
  - Electromagnetic interference (EMI) from electronics caused by Electromagnetic Fields (EMF).
  - USB 3.0 ports and computer peripherals are known culprits of EMI/EMF disrupting Zigbee.
  - 2.4 GHz Radio Frequency Interference (RFI) from Wi-Fi Routers and Wi-Fi Access Points, or others.
  - Radio proximity as well conflicting radio frequencies (Wi-Fi) overlapping used Zigbee frequency.
- Too few or far distance between Zigbee devices in Zigbee network mesh:
  - Too many obstacles between the Zigbee Coordinator or Zigbee Router and Zigbee End Device.
    - Zigbee network mesh depends on Zigbee Router devices to extend range and coverage.
    - All individual Zigbee devices do have poor signal penetration and not long range.
  - Too few Zigbee Router devices Zigbee Coordinator or Zigbee Router and Zigbee End Device.
    - Only mains-powered devices can be repeaters that extend network range and coverage.
    - Any battery-operated products are a Zigbee End Device and will never be a Zigbee Router.

#### Simple actions that should improve most Zigbee setups

- Zigbee Coordinator adapter hardware:
  - Try different physical placement and orientations of Zigbee Coordinator or its antenna.
    - Optimal placement of Zigbee Coordinator is as much in middle of the house as possible.
    - Try to place the Zigbee coordinator at some distance way from walls, ceilings and floors.
    - Try different orientations of your Zigbee antenna (or whole Zigbee coordinator adapter).
  - Update to a later released Zigbee Coordinator firmware on the Zigbee Coordinator adapter.
    - Updating FW is easy and community often provide newer firmware than manufacturer.
  - Buy a more powerful Zigbee Coordinator adapter based on a newer/modern Zigbee chip hardware.
    - Recommended as older adapter might work but have obsolete hardware and/or firmware.
- Zigbee Coordinator and Zigbee devices RFI sensitive as susceptible to all types of EMI/EMF interference:
  - Use a long USB extension cable to place Zigbee Coordinator away from interference and obstacles.
    - Try to make sure the USB extension cable has proper shielded (thick cables usually do).
    - Longer USB extension cable also make it easier to orient Zigbee Coordinator and antenna.
  - Only connect Zigbee Coordinator USB adapter to a USB 2.0 ports (or a powered USB 2.0 hub).
    - USB 3.0 ports and peripherals are infamously known for causing EMI/EMF interference.
  - Shield close unshielded computers and unshielded peripherals by using metal enclosures/chassis.
    - Single-board-computers and USB hard drives are especially known as source of EMI/EMF.
  - Change Wi-Fi Router and Access Points channels to not interfere with the default Zigbee channel.
    - Changing Wi-Fi channel on Wi-Fi devices is recommended over changing Zigbee channel.
  - Change Zigbee channel on Zigbee network if have too many strong sources of overlapping Wi-Fi.
    - Changing Zigbee channel is considered a last resort so make sure to do research before.
      - Normally require re-pairing all, and not all devices support all Zigbee channels.
- Too few or far distance between Zigbee devices in Zigbee network mesh:
  - Zigbee network mesh depends on having Zigbee Router devices to extend range and coverage:
    - Add additional mains-powered Zigbee devices known to be good as Zigbee Router device.
      - Add more Zigbee Router devices to extend network mesh range and coverage.
      - Note that not all mains-powered Zigbee devices act as a Zigbee Router device.
      - Some Zigbee Router devices are know to only work well with same brand devices.  
    - Buy a few known good dedicated Zigbee Router product (ex. “IKEA Tradfri Signal Repeater”).
      - Search community forums for "Zigbee signal repeater" or "Zigbee range extender".
    - Buy a few extra new Zigbee USB adapters to use after flashing with Zigbee Router firmware.
    - Reflash/reuse Zigbee USB adapters to Zigbee Router when upgrading Zigbee Coordinator.

#### External resources for initial Zigbee setup optimizations:

- [ZigBee and Wi-Fi Coexistence (Metageek - Advanced Wi-Fi Lessons and Training)](https://community.home-assistant.io/t/advice-on-zigbee-range-extending/175882/)
- [Choose your Zigbee channel wisely (Liam Alexander Colman - Home Assistant Guide)](https://home-assistant-guide.com/2020/10/29/choose-your-zigbee-channel-wisely/)
- [Smart Home Protocols: ZigBee Explained (Everything Smart Home on YouTube)](https://www.youtube.com/watch?v=UmpDXc3cXbU&ab_channel=EverythingSmartHome)
- [Improve Zigbee network performance and stability with these 3 tips (Home Automation Guy on YouTube)](https://www.youtube.com/watch?v=t-gw7kURXCk&t=2s&ab_channel=HomeAutomationGuy)
- [Zigbee 2022 TIPS with Home Assistant (digiblurDIY on YouTube)](https://www.youtube.com/watch?v=1dcAXkJxzcY&ab_channel=digiblurDIY)
- [How to improve your Zigbee network (Gadget-Freak)](https://gadget-freakz.com/how-to-improve-your-zigbee-network/)
- [Top 5 Reasons for Failures in ZigBee Deployments (NTS)](https://nts.com/services/education/zigbee-best-practices/)
- [Electrolama ZZH Adapter Troubleshooting and FAQ](https://electrolama.com/radio-docs/troubleshooting/)
- [2.4GHz Intra-System (or Self/Platform) Interference Demonstration (Unit 3 Compliance Blog)](https://www.unit3compliance.co.uk/2-4ghz-intra-system-or-self-platform-interference-demonstration/)
- [USB 3.0 Radio Frequency Interference Impact on 2.4 GHz Wireless Devices (Intel White Paper)](https://www.usb.org/sites/default/files/327216.pdf)

kroimon commented:
LGTM.

I like the bullet point style. I don't think it can be any shorter without leaving out important hints or information.

Hedda committed:
Merge branch 'current' into patch-49

Hedda committed:
Fix linttent error for “Zigbee” instead of “ZigBee"

Fix lint text error for incorrect usage of the term: “ZigBee”, use “Zigbee” instead  terminology

Hedda committed:
Merge branch 'current' into patch-49

Hedda committed:
Merge branch 'current' into patch-49

Hedda commented:
Any chance for a review of the rewrite? -> https://github.com/home-assistant/home-assistant.io/pull/18864/commits/970295a277e8f01d3ee39eeeaacf453625b988d3

Hedda commented:
Not stale, still waiting for new review. Almost 1-year PR anniversery soon.

Hedda commented:
Can we please merge this as-is now and improve the content later as feel it is more important to have something than nothing?

In related news to that, I just watched the Home Assistant 2022.8 Release Party, and in it some of this was brought up, specifically, it was discussed that users should really use a USB extension cable and keep radios away from USB 3.x devices, see:  

https://youtu.be/m9gKFH8WlzY?t=6239 (starting at time 1:44:00 into that video).

I also read that use of an USB extension cable and interference has now been highlighted by the new Bluetooth integration as well:

https://www.home-assistant.io/integrations/bluetooth#installing-a-usb-bluetooth-adapter

"_If you experience an unreliable Bluetooth connection, installing a short USB extension cable between your Bluetooth adapter and your Home Assistant server may improve reliability._"

https://www.home-assistant.io/integrations/bluetooth#bluetooth-interference-with-other-devices

**_Bluetooth interference with other devices_**

**_Devices that are using the 2.4 GHz band, like Wi-Fi, Zigbee, and USB3 devices (and their cable connections) are known to affect Bluetooth reception. Especially external SSD drives with USB3 cables are known to block the Bluetooth signal. Also, metal casings can decrease the Bluetooth performance of internal Bluetooth Adapters._**

**_The following tips may improve reception of the Bluetooth Adapter:_**

**_Try to place USB3 devices (SSD, etc.) as far away as possible from your Bluetooth Adapter, e.g., by using an extension cable.
Use a USB3 extension cable with proper shielding and ferrite clamps.
Use a (good quality) external Bluetooth adapter with an antenna._**

Hedda commented:
FYI, my related/respective PR for the Bluetooth integration documentation about guidance for dealing with similar sources of interference for Bluetooth has now been approved and merged after making changes as suggested in reviews:

https://github.com/home-assistant/home-assistant.io/pull/23688

https://www.home-assistant.io/integrations/bluetooth/#bluetooth-interference-with-other-devices

### Bluetooth interference with other devices

Sources of interference for radios can lead to transmission/reception loss or connection problems and show symptoms such as errors/failures when sending and receiving Bluetooth messages that can cause significant degradation in performance. Below are some basic but essential tips for getting a good setup starting point to achieve better signal quality, coverage, and extended range.

Following all these optimization tips below should significantly improve the reception of your Bluetooth radio adapter. The below insights describe working around the well-known limitations of low-power 2.4 GHz digital radio. It can resolve or avoid many known issues caused by interference or poor placement of your Bluetooth radio adapter or devices.

Computers, peripherals, and devices generate [electromagnetic interference (also known as EMI/EMI/RMI)](https://en.wikipedia.org/wiki/Electromagnetic_interference), which can interfere with signals transmissions on the 2.4 GHz radio band frequency, and degrade the wireless communication with your Bluetooth adapter/devices.

For example, unshielded USB 3 port and their cables are especially infamously known to affect 2.4 GHz radio reception. Place your Bluetooth adapter far away as possible from any potential sources of EMI/EMI/RMI by using a long, adequately shielded USB extension cable.

#### Simple actions that should improve most Bluetooth setups and common root causes of interference

- Bluetooth adapter hardware:
  - Bad performance from old/outdated Bluetooth adapter hardware or poor Bluetooth adapter antenna:
    - Buy and use a supported Bluetooth USB adapter based on newer/modern chip hardware.
      - Consider a Bluetooth adapter that has an external antenna.
      - While older adapters might work, they could have obsolete hardware or old firmware, which prevents reliable operation.
  - Poor or outdated Bluetooth adapter firmware on the Bluetooth adapter:
    - Update to the latest Bluetooth chip firmware on the Bluetooth adapter. Updating firmware is usually straightforward if the manufacturer or the chip maker provides one.
- Bluetooth adapters are RFI sensitive and can be very susceptible to all types of EMI/EMF interference:
  - Poor placement of the Bluetooth adapter or wrong orientation of Bluetooth adapter antenna:
    - Use a long USB extension cable to place the Bluetooth adapter away from interference and obstacles.
      - Ensure the USB extension cable is adequately shielded (thick cables usually have this).
        - A USB extension cable makes orienting the Bluetooth adapter/antenna easier.
    - Try different physical placement and orientations of the Bluetooth adapter or its antenna:
      - The optimal placement of the Bluetooth adapter is close to the middle of the house as possible.
      - Try to place the Bluetooth adapter at some distance away from walls, ceilings, and floors.
      - Try different orientations of the adapter's external antenna (or whole Bluetooth adapter).
  - USB 3.0 ports/computers/peripherals are known culprits of RFI/EMI/EMF disruption. (See Ref. [1](https://www.usb.org/sites/default/files/327216.pdf) and [2](https://www.unit3compliance.co.uk/2-4ghz-intra-system-or-self-platform-interference-demonstration/)).
    - Make sure to only connect the Bluetooth USB adapter to a USB 2.0 port (and not to a USB 3.x port). 
      - If your computer only has a USB 3.x port then connect the adapter via a powered USB 2.0 hub:
        - A USB 2.0 hub will convert USB 3.0 to a USB 2.0 port and avoid USB 3.0 EMF.
          - A USB 2.0 hub that uses an external power supply will ensure power requirements are fulfilled.
    - Shield any unshielded computers/peripherals/devices by adding all-metal enclosures/chassis/casings.
      - Single-board-computers and USB 3.x hard drives are especially known as source of EMF/EMI/RFI.
        - Be aware metal casings can decrease the performance of internal/built-in Bluetooth adapters.
      - Also, be sure to use adequately shielded USB cables for any such peripherals/devices too.
  - 2.4 GHz RF Interference (RFI) from Wi-Fi Routers and Wi-Fi Access Points or other devices:
    - While Bluetooth is designed to coexist with Wi-Fi, its stronger signal can interfere.
      - To play it safe, try to place your Bluetooth adapter away from Wi-Fi access points.
    - Place Bluetooth adapters far away from electrical/power wires/cables, power supplies, and household appliances.

Hedda commented:
Rewrote the whole thing for a new PR in the hopes that it will be reviewed -> https://github.com/home-assistant/home-assistant.io/pull/24432

Hedda commented:
Closing in favor of https://github.com/home-assistant/home-assistant.io/pull/24432

Hedda committed:
Merge branch 'current' into patch-49

Hedda committed:
Rewritten as per feedback

Rewritten as per feedback

Hedda committed:
Merge branch 'current' into patch-49

frenck commented on source/_integrations/zha.markdown:
```diff
@@ -432,6 +432,60 @@ Such a custom "ZHA Device Handler" are Python scripts that internally are also r
+
+For example, interference from USB 3.0 ports and unshielded USB peripheral cables are especially infamously known to affect 2.4 GHz radio reception for low-power/low-bandwidth devices. Therefore you should always place your Zigbee adapter far away as possible from any potential sources of EMI/EMI/RMI, preferably by using an adequately long shielded USB extension cable connected to a USB 2.0 port.
+
+Zigbee also uses [mesh networking topology](https://en.wikipedia.org/wiki/Mesh_networking), which means that most mains-powered devices are a "Zigbee Router" that can act as a signal repeater and range extended by transmitting data over long distances by passing data messages through the Zigbee network mesh of intermediate devices to reach more distant Zigbee devices. Thus to have a healthy Zigbe ...
```
```suggestion
Zigbee also uses [mesh networking topology](https://en.wikipedia.org/wiki/Mesh_networking), which means that most mains-powered devices are a "Zigbee Router" that can act as a signal repeater and range extended by transmitting data over long distances by passing data messages through the Zigbee network mesh of intermediate devices to reach more distant Zigbee devices. Thus to have a healthy Zigbee network, you need many Zigbee Router devices relatively close to each other in order to achieve good coverage and range.
```

frenck commented on source/_integrations/zha.markdown:
```diff
@@ -432,6 +432,60 @@ Such a custom "ZHA Device Handler" are Python scripts that internally are also r
+
+Zigbee also uses [mesh networking topology](https://en.wikipedia.org/wiki/Mesh_networking), which means that most mains-powered devices are a "Zigbee Router" that can act as a signal repeater and range extended by transmitting data over long distances by passing data messages through the Zigbee network mesh of intermediate devices to reach more distant Zigbee devices. Thus to have a healthy Zigbe ...
+
+#### Simple actions that should improve most Zigbee setups and common root causes of interference
```
```suggestion
#### Actions that can improve most Zigbee setups and common root causes of interference
```

frenck commented on source/_integrations/zha.markdown:
```diff
@@ -432,6 +432,60 @@ Such a custom "ZHA Device Handler" are Python scripts that internally are also r
+
+- Zigbee adapter hardware:
+  - Bad performance from old/outdated/obsolete Zigbee adapter hardware or poor Zigbee adapter antenna:
+    - Buy and use a supported Zigbee USB adapter based on newer/modern chip hardware.
+      - Consider a Zigbee adapter that has an external antenna.
+      - While older adapters might work, they could have obsolete hardware or old firmware, which prevents reliable operation.
+  - Poor or outdated Zigbee adapter firmware on the Zigbee adapter:
+    - Update to a later Zigbee chip firmware on the Zigbee adapter. Updating firmware is usually straightforward if the manufacturer or the chip maker provides one.
+- Zigbee adapters are RFI sensitive and can be very susceptible to all types of EMI/EMF interference:
+  - Poor placement of the Zigbee adapter or wrong orientation of the Zigbee adapter antenna:
+    - Use a long USB extension cable to place the Zigbee adapter away from interference and obstacles.
+      - Ensure the USB extension cable is adequately shielded (thick cables usually have this).
+        - A USB extension cable makes orienting the Zigbee adapter/antenna easier.
+    - Try different physical placement and orientations of the Zigbee adapter or its antenna:
+      - The optimal placement of the Zigbee adapter is close to the middle of the house as possible.
+      - Try to place the Zigbee adapter at some distance away from walls, ceilings, and floors.
+      - Try different orientations of the adapter's external antenna (or the whole Zigbee adapter).
+  - USB 3.0 ports/computers/peripherals are known culprits of RFI/EMI/EMF disruption. (See Ref. [1](https://www.usb.org/sites/default/files/327216.pdf) and [2](https://www.unit3compliance.co.uk/2-4ghz-intra-system-or-self-platform-interference-demonstration/)).
+    - Make sure to only connect the Zigbee USB adapter to a USB 2.0 port (and not to a USB 3.x port). 
+      - If your computer only has a USB 3.x port then buy and connect the adapter via a powered USB 2.0 hub:
+        - Using via USB 2.0 hub will in practice convert USB 3.0 to a USB 2.0 port and thus avoid USB 3.0 EMF.
+          - A USB 2.0 hub that uses an external power supply will ensure power requirements are fulfilled.
+    - Shield any unshielded computers/peripherals/devices by adding all-metal enclosures/chassis/casings.
+      - Single-board-computers and USB 3.x hard drives are especially known as sources of EMF/EMI/RFI.
+        - Be aware metal casings can decrease the performance of internal/built-in Zigbee adapters.
+      - Also, be sure to use adequately shielded USB cables for any such peripherals/devices too.
+  - 2.4 GHz RF Interference (RFI) from Wi-Fi Routers and Wi-Fi Access Points or other devices:
+    - First of all, try to place your Zigbee adapter away from Wi-Fi access points or sources of WiFi.
+    - Place Zigbee adapters far away from electrical/power wires/cables, power supplies, and household appliances.
+    - Zigbee could have overlapping frequency ranges with Wi-Fi, see the section above on defining Zigbee channel use.
+  - Add more and decrease the distance between Zigbee devices in Zigbee network mesh to get better range and coverage:
+    - Zigbee uses [mesh networking](https://en.wikipedia.org/wiki/Mesh_networking) and depends on having many "Zigbee Router" devices to extend range and coverage:
+      - Recommendation is to add additional mains-powered Zigbee devices known to be good Zigbee Router devices.
+        - Add more Zigbee Router devices and reduce their distances to extend network mesh coverage and range.
+          - Note that not all mains-powered devices have firmware that makes them act as a Zigbee Router device.
+          - Some brands/models of Zigbee Router devices are known to only work well with the same brand of devices.
+      - Buy a few known good dedicated Zigbee Router products (for example the “[IKEA Tradfri Signal Repeater](https://www.google.com/search?q=IKEA+Tradfri+Signal+Repeater)”).
+        - Search community forums for more "Zigbee signal repeater" or "Zigbee range extender" Zigbee Router tips.
+      - Buy a few additional new Zigbee USB adapters to use after re-flashing them with Zigbee Router firmware.
+        - Reflash/reuse modern Zigbee USB adapters to act as Zigbee Router devices by changing Zigbee firmware.
```
This really doesn't work as a big list. For example:

![image](https://user-images.githubusercontent.com/195327/212767828-6e87ff4d-f517-400c-8b4f-fc89db894f86.png)

Not even to start reading this one mobile 🙈  This needs to be structured differently.

Hedda replied:
Got any specific suggestions on how to make it better? That is the same format that got accepted for Bluetooth https://www.home-assistant.io/integrations/bluetooth/#bluetooth-interference-with-other-devices

Hedda replied:
> Not even to start reading this one mobile 🙈 This needs to be structured differently.

Is that really a showstopper? It is not better to now just push this into the ZHA documentation as-is and then improve it?

I think that some documentation about this very important subject of interference avoidance must be better than none.

Hedda replied:
Tried again to restructure that last segment as best I could.

Hedda committed:
Update source/_integrations/zha.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

Hedda committed:
Update source/_integrations/zha.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

frenck commented:
I've marked this PR, as changes are requested that need to be processed.
Please un-draft it once it is ready for review again by clicking the "Ready for review" button.

Thanks! 👍 

../Frenck

Hedda commented:
Does anyone else have any constructive feedback on how to rewrite this to make it good enough for merging into the ZHA docs?

Hedda committed:
Merge branch 'current' into patch-49

Hedda committed:
Restructured last segments

frenck commented:
The constructive feedback was given: it is a huge unreadable bullet list, that is hard to read, especially on mobile.

The solution is: don't make a huge bullet list. Make some chapters/paragraphs with some readable text in it that flows and is easy to the eyes to read. Sure some sections may have some bullet point, that is fine. The huge list is the issue here.

../Frenck

Hedda committed:
Shorten, restructured and rewritten bullets

Shorten, restructured and rewritten the bullet point list

Hedda commented:
> The solution is: don't make a huge bullet list. Make some chapters/paragraphs with some readable text in it that flows and is easy to the eyes to read. Sure some sections may have some bullet point, that is fine. The huge list is the issue here.

OK, based on that more specific feedback I have now tried to clean and restructured as separate smaller bullet point lists, removing some parts as well as splitting some other parts into shorter chapters/paragraphs instead.

Hedda commented:
For reference, Home Assistant SkyConnect website and FAQ on webpage where to buy it now contain some of this information:

https://skyconnect.home-assistant.io/connectivity/

https://www.home-assistant.io/skyconnect -> FAQs -> "_Why include a USB extension cable?_"

  * "_USB 3.0 ports (the ones with blue on the inside) are known to cause significant noise and radio interference to any 2.4Ghz wireless devices. This includes Zigbee and Thread. If you do not use the extension cable, it may not work at all, and if it does, it could be flaky at best with intermittent problems (issues with pairing, device dropouts, unreachable devices, timeout errors, etc)._"

Issue with USB 3.0 interference for Zigbee was also mentioned in Home Assistant's latest newsletter for Building the Open Home:


https://building.open-home.io/assisting-the-open-home/

### _Zigbee interference caused by USB 3.0_

_A lot of users build their smart home around a Raspberry Pi. It's quite popular to expand the storage using an external hard drive connected to the Raspberry Pi via USB 3. What most users don't know is that USB 3 causes enormous interference with any Zigbee device or stick that comes close, to a point that it no longer works._

_Gabriela recently joined the Nabu Casa team as a technical documentation writer and has been focusing on documentation to help users avoid common pitfalls like these. She updated the [documentation for the Home Assistant SkyConnect](https://skyconnect.home-assistant.io/connectivity/?ref=building-the-open-home) including this 30 second video that shows how bad the issue is:_

https://www.youtube.com/watch?v=tHqZhNcFEvA&t=1s&ab_channel=HomeAssistant

frenck commented on source/_integrations/zha.markdown:
```diff
@@ -437,7 +437,55 @@ Home Assistant's ZHA integration already supports most standard device types out
+Since all Zigbee Coordinator radio adapters are very sensitive/susceptible to all types of EMI/EMF/RMI you should always try to optimize the placement of the Zigbee Coordinator and avoid known sources of interference.
+
+- Use a long USB extension cable and place Zigbee Coordinator away from interference and obstacles.
+  - Ensure USB extension cable is adequately shielded (thicker cables usually have better shielding).
```
```suggestion
  - Ensure the USB extension cable is adequately shielded (thicker cables usually have better shielding).
```

frenck commented on source/_integrations/zha.markdown:
```diff
@@ -437,7 +437,55 @@ Home Assistant's ZHA integration already supports most standard device types out
+
+- Avoid USB 3.0 ports/computers/peripherals as they are known culprits of RFI/EMI/EMF disruption. (See Ref. [1](https://www.usb.org/sites/default/files/327216.pdf) and [2](https://www.unit3compliance.co.uk/2-4ghz-intra-system-or-self-platform-interference-demonstration/)).
+  - Make sure to only connect the Zigbee USB adapter to a USB 2.0 port (and not to a USB 3.x port). 
+  - If computer only has USB 3.x ports then buy and connect Zigbee Coordinator via a powered USB 2.0 hub.
```
```suggestion
  - If a computer only has USB 3.x ports then buy and connect Zigbee Coordinator via a powered USB 2.0 hub.
```

frenck commented on source/_integrations/zha.markdown:
```diff
@@ -437,7 +437,55 @@ Home Assistant's ZHA integration already supports most standard device types out
+
+- Shield any unshielded computers/peripherals/devices by adding all-metal enclosures/chassis/casings.
+  - Use shielded USB cables for all external peripherals/devices, especially USB 3.x peripherals.
+  - Be aware that metal casings can decrease performance of an internal/built-in Zigbee Coordinator.
```
```suggestion
  - Be aware that metal casings can decrease the performance of an internal/built-in Zigbee Coordinator.
```

frenck commented on source/_integrations/zha.markdown:
```diff
@@ -437,7 +437,55 @@ Home Assistant's ZHA integration already supports most standard device types out
+  - Use shielded USB cables for all external peripherals/devices, especially USB 3.x peripherals.
+  - Be aware that metal casings can decrease performance of an internal/built-in Zigbee Coordinator.
+
+- Avoid Wi-Fi Routers and Wi-Fi Access Points, alternatively change WiFi channel or Zigbee channel.
+  - Place your Zigbee Coordinator away from any Wi-Fi access points and all other sources of WiFi.
+  - Wi-Fi frequency ranges can overlap with Zigbee, see section above on defining Zigbee channel use.
```
```suggestion
- Avoid Wi-Fi Routers and Wi-Fi Access Points, alternatively change the Wi-Fi channel or Zigbee channel.
  - Place your Zigbee Coordinator away from any Wi-Fi access points and all other sources of WiFi.
  - Wi-Fi frequency ranges can overlap with Zigbee, see the section above on defining Zigbee channel use.
```

frenck reviewed (changes requested):
This looks nice now! Good work.
Some nitpicks left, which I provided suggestions for.

Hedda committed:
Update source/_integrations/zha.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

Hedda committed:
Update source/_integrations/zha.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

Hedda committed:
Update source/_integrations/zha.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

Hedda committed:
Update source/_integrations/zha.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

Hedda committed:
Merge branch 'current' into patch-49

Hedda commented:
OK marking this as ready for review now.

frenck reviewed (approved):
Thanks, @Hedda 👍  🎉 

../Frenck
